torch/distributed/distributed_c10d.py

```diff
@@ -75,8 +76,22 @@ def __new__(cls, name):
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
             raise ValueError("Invalid backend: '{}'".format(name))
+        elif value != Backend.GLOO and value != Backend.NCCL and value != Backend.MPI:
+            value = name
         return value
 
+    @classmethod
+    def register_backend(cls, name, func):
+        """Registers a new backend.
+
+        This class method is used by 3rd party cpp extension to register new backend.
+
+        Arguments:
+            name (str): Backend name matching with the one in `init_process_group()`.
+            func (function): Function handler that instantiates the backend.
```

Comment:
As we expect the function to take certain arguments (`prefix_store`, `rank`, `world_size`, and `timeout`), can we explain those args here as well? 

Ideally, we might want to make this function first class citizen as well with clear signature and documents. But as we can mark this feature as experimental, we should be able to have some room to change it in the future when necessary.

Reply:
will add comments here